File: solutions/elixir/lasagna/2/lib/lasagna.ex
defmodule Lasagna do
  # Please define the 'expected_minutes_in_oven/0' function
  def expected_minutes_in_oven do
    40
  end;

  # Please define the 'remaining_minutes_in_oven/1' function
  def remaining_minutes_in_oven(minutes_running) do
    expected_time = expected_minutes_in_oven();
    expected_time-minutes_running
  end;

  # Please define the 'preparation_time_in_minutes/1' function
  def preparation_time_in_minutes(n_layers) do
    n_layers*2
  end;

  # Please define the 'total_time_in_minutes/2' function
  def total_time_in_minutes(n_layers, nMinutes) do
    nTimeLayers = preparation_time_in_minutes(n_layers);
    nMinutes+nTimeLayers
  end;

  # Please define the 'alarm/0' function
  def alarm do
    "Ding!"
  end;
end

File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    // TODO: define the 'expectedMinutesInOven()' method
    public int expectedMinutesInOven(){
        return 40;
    }
    // TODO: define the 'remainingMinutesInOven()' method
    public int remainingMinutesInOven(int n){
        return 40 - n;
    }
    // TODO: define the 'preparationTimeInMinutes()' method
    public int preparationTimeInMinutes(int n){
        return n * 2;
    }

    // TODO: define the 'totalTimeInMinutes()' method
    public int totalTimeInMinutes(int n, int n2){
        return (n * 2) + n2;
    }
}
